src/client.ts

```diff
@@ -232,7 +240,7 @@ export class Client {
    *
    * @return {Promise<Response>}
    */
-  public async search(params: Query | QueryParams): Promise<Response | DoofinderResult> {
+  public async search(params: Query | SearchParams): Promise<SearchResponse> {
```

Comment:
It may be that allowing a SearchParams as params in the search method it's not necessary and adds more complexity. The programmer always should use the Query instance.

Reply:
@sonic182, @resetreboot  what do you think?

Reply:
Well, for the user of this call, it means creating a Query instance always. If you are just doing a quick search and not modifying the query afterwards, it makes it cumbersome. 

Btw, the complexity boils down to check what comes in and create a Query from the SearchParams, which I don't think it's too much complexity.

Reply:
El usuario debería conocer que existe un objeto query en el que se define la consulta a realizar. Y el cliente usa ese objeto para hacer la consulta. Eso hasta ahí me parece simple. 

Decir que el cliente puede recibir además de la Query un objeto que implemente QueryParams y que entonces también permite hacer la query me parece menos simple.

Reply:
I believe that is not the case, as the Query properties match the QueryParams object. In the end, both are objects, but one sports some sort of methods. If you don't allow to send the object as-is, you're blocking a possible way for the user of this library to create their own object and access a bit of a more low level to the search parameters, making them go full on calling a fetch API and ditching our library if it comes to it. 

I say this is a library, and in the spirit of Python, we'd rather take the complexity ourselves as we are making our API easier to use than a straight HTTP query, we shouldn't get in the way to access a bit more in depth, but help when there's no need for that.

Reply:
@hailem the problem may be that we use `Query` with the provided params and `query.dump(true)` to validate them instead of using the search params directly if they're not an instance of `Query`…

Reply:
In this case the client does only provide methods to reach the endpoints, but should not enforce the use of an auxiliary class. That's why it both accepts a plain object with search params and an instance of `Query`.

Reply:
@hailem so…?